core/src/main/java/org/codice/imaging/nitf/core/tre/impl/TreParser.java

```diff
@@ -658,10 +663,10 @@ private void checkTreLocationMatchesTreSource(final String location, final TreSo
             if ((!source.equals(TreSource.UserDefinedHeaderData)) && (!source.equals(TreSource.ExtendedHeaderData))) {
                 throw new NitfFormatException("TRE is only permitted in a file-level header, or in an overflow DES");
             }
-        } else if (location.equalsIgnoreCase("image")) {
-            if ((!source.equals(TreSource.ImageExtendedSubheaderData)) && (!source.equals(TreSource.UserDefinedImageData))) {
-                throw new NitfFormatException("TRE is only permitted in an image-related sub-header, or in an overflow DES");
-            }
+        } else if (location.equalsIgnoreCase("image")
+                && (!source.equals(TreSource.ImageExtendedSubheaderData))
+                && (!source.equals(TreSource.UserDefinedImageData))) {
```

Comment:
Probably not worth refactoring, but when I see these kinds of tests on the String "location", it makes me think that Location should be an object with methods like "isFile", "isImage", "isExtendedSubheaderData", etc. As well as a public method "matchesSource(TreSource source)"

Reply:
Sounds good to me, but I will leave that for a future improvement for now.